src/main/java/seedu/toluist/model/Task.java

```diff
@@ -272,16 +284,118 @@ public void setTaskPriority(TaskPriority priority) {
     }
 
     public void setTaskPriority(String priorityString) {
-        if (priorityString.equalsIgnoreCase(HIGH_PRIORITY_STRING)) {
+        switch (priorityString.toLowerCase()) {
+        case HIGH_PRIORITY_STRING:
             setTaskPriority(TaskPriority.HIGH);
-        } else if (priorityString.equalsIgnoreCase(LOW_PRIORITY_STRING)) {
+            break;
+        case LOW_PRIORITY_STRING:
             setTaskPriority(TaskPriority.LOW);
-        } else {
+            break;
+        default:
             throw new IllegalArgumentException("Task priority must be either 'low' or 'high'.");
         }
     }
 
     public LocalDateTime getCompletionDateTime() {
         return completionDateTime;
     }
+
+    //@@author A0127545A
```

Comment:
Maybe RecurringTask should be another subclass.  This class file is getting quite huge